scintillometry/io/psrfits/hdu.py

```diff
@@ -67,17 +67,43 @@ def location(self):
         except KeyError:
             return None
 
+    @location.setter
+    def location(self, loc):
+        """ Location setter. input should be an Astropy EarthLocation"""
+        # TODO, for the space base observatory, Earth Location may not apply.
+        self.hdu.header['ANT_X'] = loc.itrs.x.to_value(u.m)
+        self.hdu.header['ANT_Y'] = loc.itrs.y.to_value(u.m)
+        self.hdu.header['ANT_Z'] = loc.itrs.z.to_value(u.m)
+
     @property
     def start_time(self):
         return (Time(self.header['STT_IMJD'], format='mjd', precision=9,
                      location=self.location) +
                 TimeDelta(self.header['STT_SMJD'], self.header['STT_OFFS'],
                           format='sec', scale='tai'))
 
+    @start_time.setter
+    def start_time(self, time):
+        """ Set the start_time, the input value should be an Time object"""
+        if not time.isscalar:
+            raise ValueError("Input time should be a scalar")
+        mjd_int = int(time.mjd)
+        mjd_frac = (time - Time(mjd_int, scale=time.scale, format=time.format))
+        mjd_frac = mjd_frac.to(u.s)
```

Comment:
Replace this line and the one below with:
```
frac_sec, int_sec = np.modf(mjd_frac.to_value(u.s))
```

Reply:
My time object does not have 'to_value()'

Reply:
OK, then I guess it will still have to be `to(u.s).value` - we can simplify it in once we only support astropy 4.0... https://github.com/astropy/astropy/pull/8762 (this is why i thought it was possible...)